x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/transport/netty4/SecurityNetty4Transport.java

```diff
@@ -167,16 +170,18 @@ protected ServerChannelInitializer getSslChannelInitializer(final String name, f
     private class SecurityClientChannelInitializer extends ClientChannelInitializer {
 
         private final boolean hostnameVerificationEnabled;
+        private final String sniServerName;
 
-        SecurityClientChannelInitializer() {
+        SecurityClientChannelInitializer(DiscoveryNode node) {
+            this.sniServerName = node.getAttributes().get("sni_server_name");
```

Comment:
I think we should always use the `node.getHostName()` as well as an attribute if it's present. I think we can use something like `alt_host_name` ie. something more generic? @jaymode WDYT?

Reply:
> I think we should always use the node.getHostName() as well as an attribute if it's present

++

> I think we can use something like alt_host_name ie. something more generic?

How about `server_names`?

Reply:
`server_names` sounds good to me

Reply:
Should it be `server_names`. or `server_name`? Since the attribute map is string to string, I think the non-plural makes more sense. But let me know and I can do whatever.

Reply:
++ plural is good